### ISSUE TITLE
Taking an opinionated approach to getting started with F#

### ISSUE THREAD

**Issue by cartermp** [fsharp/fssf-ask-the-board#1]
This pertains to the FSSF site, which is usually the first one people go to when they're learning F# for the first time.

I believe that one of the bigger blockers to F# (and .NET) adoption is twofold:

1. There are many ways to get started.
2. It's not clear which is the best if you're a newcomer to F# (or .NET).

I think the best way to solve this is to take an opinionated stance on what people should use to get started, then link to a page which offers all the other alternatives.

What does the board think?

**Comment by realvictorprm** [comment]
Sounds right to me. This may be provided through an code runner on the main page I think (e.g. First sample code should point people to a quick tutorial maybe)

Congrats, the first issue @cartermp :smile:

**Comment by julienXX** [comment]
I started playing with F# over a month ago and one thing that really needs clarification for me is .NET vs Mono vs .NET Core and the associated tooling. It's not F# "core" stuff but it would be useful to get people started.

**Comment by lambdakris** [comment]
Not to discourage this notion but I think there is a snag that will crop up any way you try to come at this which is why it has not emerged on it's own as yet. Let's say you want to start with .NET Core CLI and VS Code + Ionide since it is the leanest and most similar setup to other langs/plats (and also only VS Code + Ionide is the only fully functioning editor for .NET Core...). But as soon as you want to bring in Ionide you need FSAC, which needs FCS, which needs .NET Fx or Mono, so you are essentially forcing the beginner to install BOTH .NET Core and .NET Fx/Mono and have some inkling of their independence from a runtime standpoint but their co-necessity from a tooling standpoint (you can try to hand wave their attention away, but something invariably goes wrong and to troubleshoot they will need to fiddle with the gory details). Now, let's say you instead go for .NET Fx/Mono and Rider or Visual Studio IDE, well, now you give up significant command line power (which will contrast starkly to other langs/plats) and neither Rider nor Visual Studio 2017 are yet fully functional (even on full .NET). And in the case of Windows different versions have different version of .NET and Windows SDK so you may have to install additional Windows SDKs and .NET Frameworks appropriate for your windows version, maybe even the Microsoft Build Tools for MSBuild.... In a way, the best way may be Mono and VS for Mac or VS Code on OS X...

In short, it will be easier to sell/explain/instruct F# setup and workflow when FCS/FSAC run on .NET Core and no thought need be given to .NET Fx or Mono...

**Comment by mrc** [comment]
You could also take multiple opinionated approaches, depending on where the starter is coming from. Maybe they want to write a web app on a Mac. Maybe they want to try F# for unit tests with an existing C# codebase on Windows. Maybe they want to do data science on Linux. All this stuff exists in terrific blog posts you could get there with a *very short* curated list of links (or asking the authors if you can republish on fsharp.org).

**Comment by cartermp** [comment]
Yes, outside of just using the .NET CLI and vim, there are additional prerequisites.  Once .NET Core 2.0 ships, there's an opportunity to do a spike and see if the various pieces of Ionide are able to run on .NET Core 2.0.

It's a very difficult problem to "solve" because there's no clear best choice.  But I believe that fsharp.org needs to be the one that pushes one of those options for people.

**Comment by Krzysztof-Cieslak** [comment]
Editor / IDE is least important part of it, most of them has really similar set of features and capabilities. What I'd rather want to see is promoting and showing this "opinionated" getting started story based around all other tooling - Paket, FAKE, Forge.

**Comment by FSSF** [comment]
This is being moved to be part of the overall website discussion: fsharp/fsfoundation#683

For details, please see fsharp/fsfoundation#680
